code/controllers/subsystem/map_gen.dm

```diff
@@ -0,0 +1,168 @@
+SUBSYSTEM_DEF(map_gen)
+	name = "Map Generation"
+	wait = 1
+	flags = SS_TICKER | SS_BACKGROUND
```

Comment:
Ticker and Background are not mutually exclusive. Ticker causes it to fire every tick in the Ticker bucket, which fires before the Standard and Background buckets.

Background subsystems are fully capable of NEVER firing if the server is under strain.